### PR TITLE
Enhance the fixer of interface-over-type-literal

### LINKED ISSUE #4795: Have interface-over-type-literal replace trailing comma delimiters with semicolons
### Feature request

**Is your feature request related to a problem? Please describe.**
The `interface-over-type-literal` rule essentially turns a type alias into an interface. As semicolons are the preferred delimiter for marking the end of a line, it's better to handle the case where the type alias to be fixed used comma as separators.

It would add clarity to the shape of the object, e.g.,
```
interface User {
  someData: {
    firstName: string,
    lastName: string,
    registered: boolean,
  };
  location: string;
  quantity: number;
}
```
instead of
```
interface User {
  someData: {
    firstName: string,
    lastName: string,
    registered: boolean,
  },
  location: string,
  quantity: number,
}
```

**Describe the solution you'd like**
Add an additional step to the fixer so that it checks if the type alias uses comma as its separator for member property. If so, replace them with semicolons.

**Describe alternatives you've considered**
Be less aggressive and add a warning instead, but I think it complicates the process. Automatically changing the separators makes more sense.

### PR DESCRIPTION
**Pull request by ychen10**
#### PR checklist

- [x] Addresses an existing issue: fixes  #4795 
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:
Added a step in the fixer so that it goes into the members to check if they carry a trailing comma. If they do, replace it with semicolon. The fixed interface now looks like
```
interface User {
  someData: {
    firstName: string,
    lastName: string,
    registered: boolean,
  };
  location: string;
  quantity: number;
}
```
instead of
```
interface User {
  someData: {
    firstName: string,
    lastName: string,
    registered: boolean,
  },
  location: string,
  quantity: number,
}
```

#### CHANGELOG.md entry:
[enhancement] `interface-over-type-literal` The fixer now makes sure that the interface uses semicolon to separate individual member declarations.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/tslint](https://github.com/palantir/tslint), @ychen10! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by ychen10** [comment]
Test failing as per #4784

**Comment by JoshuaKGoldberg** [comment]
Declined per #4795 😢.

In general, until an issue is marked as [Status: Accepting PRs](https://github.com/palantir/tslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22Status%3A+Accepting+PRs%22), it's better to hold off on writing a PR. You never know when an issue will be declined by some annoying maintainer.

File: src/rules/interfaceOverTypeLiteralRule.ts [hunk 1]
```diff
@@ -57,6 +57,17 @@ function walk(ctx: Lint.WalkContext): void {
             if (ctx.sourceFile.text[node.end - 1] === ";") {
                 fix.push(Lint.Replacement.deleteText(node.end - 1, 1));
             }
+            // replace trailing comma with semicolon
+            const members = node.type.members;
+            if (members.length !== 0) {
+                members.forEach(mem => {
+                    const memText: string = mem.getFullText();
+                    if (memText[memText.length - 1] === ",") {
```
Comment (JoshuaKGoldberg) [inline src/rules/interfaceOverTypeLiteralRule.ts]: If you fork this rule and use it in your own project, you might want to write some test cases around comments and spacing:

```ts
type Wat = {
    someMember : string /* gotcha! */,
}
```

**Comment by JoshuaKGoldberg** [comment]
FWIW, you can always fork this rule (copy & paste the `.js` output into a new file) and use it in your own project instead of the core rule.

File: src/rules/interfaceOverTypeLiteralRule.ts (hunk 1 above)
Comment (ychen10) [reply src/rules/interfaceOverTypeLiteralRule.ts]: Got it. Thanks!

**Pull request closed without merging**